tests/nlu/featurizers/test_lm_featurizer.py

```diff
@@ -366,342 +375,30 @@ def test_attention_mask(
-                    "is",
-                    "the",
-                    "sentence",
-                    "i",
-                    "want",
-                    "em",
-                    "bed",
-                    "ding",
-                    "s",
-                    "for",
-                ],
-            ],
-            [
-                [(0, 4), (5, 12)],
-                [(0, 3), (4, 6)],
-                [(0, 1), (3, 4), (6, 7)],
-                [(0, 4), (5, 6), (7, 11)],
-                [
-                    (0, 4),
-                    (5, 7),
-                    (8, 11),
-                    (12, 20),
-                    (21, 22),
-                    (23, 27),
-                    (28, 30),
-                    (30, 33),
-                    (33, 37),
-                    (37, 38),
-                    (39, 42),
-                ],
-            ],
-        ),
-        (
-            "roberta",
-            None,
-            [
-                "Good evening.",
-                "hello",
-                "you're",
-                "r. n. b.",
-                "rock & roll",
-                "here is the sentence I want embeddings for.",
-            ],
-            [
-                ["Good", "even", "ing"],
-                ["hello"],
-                ["you", "re"],
-                ["r", "n", "b"],
-                ["rock", "&", "roll"],
-                [
-                    "here",
-                    "is",
-                    "the",
-                    "sent",
-                    "ence",
-                    "I",
-                    "want",
-                    "embed",
-                    "d",
-                    "ings",
-                    "for",
-                ],
-            ],
-            [
-                [(0, 4), (5, 9), (9, 12)],
-                [(0, 5)],
-                [(0, 3), (4, 6)],
-                [(0, 1), (3, 4), (6, 7)],
-                [(0, 4), (5, 6), (7, 11)],
-                [
-                    (0, 4),
-                    (5, 7),
-                    (8, 11),
-                    (12, 16),
-                    (16, 20),
-                    (21, 22),
-                    (23, 27),
-                    (28, 33),
-                    (33, 34),
-                    (34, 38),
-                    (39, 42),
-                ],
-            ],
-        ),
+        ("bert", "sentence embeddings", [1, 2]),
+        ("bert", "this is a test", [1, 1, 1, 1]),
+        ("gpt", "sentence embeddings", [1, 2]),
+        ("gpt", "this is a test", [1, 1, 1, 1]),
+        ("gpt2", "sentence embeddings", [2, 3]),
+        ("gpt2", "this is a test", [1, 1, 1, 1]),
+        ("xlnet", "sentence embeddings", [1, 3]),
+        ("xlnet", "this is a test", [1, 1, 1, 1]),
+        ("distilbert", "sentence embeddings", [1, 4]),
+        ("distilbert", "this is a test", [1, 1, 1, 1]),
+        ("roberta", "sentence embeddings", [2, 3]),
+        ("roberta", "this is a test", [1, 1, 1, 1]),
```

Comment:
Oh! I think this was a misunderstanding on my part -- I removed `test_lm_featurizer_edge_cases` because I thought you said it should be removed in the original problem description:

> Now that our tokenizers don't split tokens into sub-tokens, the test in (1) is redundant.

I thought that was weird but did it anyways.. I should've checked with you, sorry.

Reply:
So I added this back, but the only thing I think we really care about is the sub-token count. Correct me if I'm wrong? Should I be checking the actual values as well? Seems like this is covered by `shape_values`, but I don't know if these edge cases can cause weird behaviour.

Reply:
I think testing for sub-token count is good enough.